### PR TITLE
feat: Adding a show all button to the column/metrics list in the explore view (Allow more than 50 columns to be shown)

### PR DESCRIPTION
**Pull request by cccs-RyanS**
### SUMMARY
This is a simple addition to the explore view. The explore view limits the amount of data / columns displayed at once to 50. We wish to have the option of showing all of the columns / metrics at one time. The look / functionality was discussed in

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before at 50 columns

before_over_50

After the changes on a datasource with over 50 columns (in various states Before at 50 columns)

less shown

showall
)

Capture

showless

After the changes on a datasource with less than 50 columns 

under_50

### TESTING INSTRUCTIONS
Open a data source with with more than 50 columns or metrics. At the button of there will be a "Show All..." button, clicking this button will cause  full list of columns / metrics to render. If all columns / metrics are rendering a "Show Less..." button will appear, clicking this will cause only 50 elements to render.

### ADDITIONAL INFORMATION
- [x] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 1]
```diff
@@ -113,6 +125,11 @@ export default function DataSourcePanel({
     columns,
     metrics,
   });
+  const [showAllMetrics, setShowAllMetrics] = useState(false);
+  const [showAllColumns, setShowAllColumns] = useState(false);
+
+  const DEFAULT_MAX_COLUMNS_LENGTH = 50;
+  const DEFAULT_MAX_METRIC_LENGTH = 50;
```
Comment (cccs-rc) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:132]: Biggest nitpick ever. 
```suggestion
  const DEFAULT_MAX_METRICS_LENGTH = 50;
```

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 2]
```diff
@@ -219,6 +240,15 @@ export default function DataSourcePanel({
                 )}
               </LabelContainer>
             ))}
+            {lists.metrics.length > DEFAULT_MAX_METRIC_LENGTH ? (
+              <ButtonContainer>
+                <Button onClick={() => setShowAllMetrics(!showAllMetrics)}>
+                  {showAllMetrics ? t('Show Less...') : t('Show all...')}
```
Comment (cccs-rc) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:246]: One more giant nitpick. 
```suggestion
                  {showAllMetrics ? t('Show less...') : t('Show all...')}
```

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 3]
```diff
@@ -113,6 +125,11 @@ export default function DataSourcePanel({
     columns,
     metrics,
   });
+  const [showAllMetrics, setShowAllMetrics] = useState(false);
+  const [showAllColumns, setShowAllColumns] = useState(false);
+
+  const DEFAULT_MAX_COLUMNS_LENGTH = 50;
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:131]: Should we put these constants in the superset config, then access these values from store of redux `explore.common.conf.DEFAULT_MAX_COLUMNS_LENGTH` and `explore.common.conf.DEFAULT_MAX_METRICS_LENGTH`

**Review by zhaoyongjie — APPROVED** [review]
Tested in local, LGTM. Thanks for your contrabution. 
cc @junlincc 

closes: https://github.com/apache/superset/issues/14751

https://user-images.githubusercontent.com/2016594/123552022-def3e180-d7a6-11eb-80da-7fa3faa9f3b9.mp4

**Review by zhaoyongjie — APPROVED** [review]

**Pull request merged**